yaml.go

```diff
@@ -96,6 +96,61 @@ func JSONToYAML(j []byte) ([]byte, error) {
 	return yaml.Marshal(jsonObj)
 }
 
+// internalYamlUnmarshal works like yaml.Unmarshal but supports streams.
+func internalYamlUnmarshal(b []byte, out interface{}) error {
+	return internalYamlUnmarshalWrapper(false, b, out)
+}
+
+// internalYamlUnmarshalStrict works like yaml.UnmarshalStrict but supports streams.
+func internalYamlUnmarshalStrict(b []byte, out interface{}) error {
+	return internalYamlUnmarshalWrapper(true, b, out)
+}
+
+// internalYamlUnmarshalWrapper is a wrapper around yaml.Decoder that allows
+// for decoding yaml streams.
+func internalYamlUnmarshalWrapper(strict bool, b []byte, out interface{}) error {
+	outV := reflect.ValueOf(out)
+	if !outV.CanSet() {
+		outV = outV.Elem()
+	}
+
+	if len(bytes.TrimSpace(b)) == 0 {
+		outV.Set(reflect.Zero(outV.Type()))
+		return nil
+	}
+
+	decoder := yaml.NewDecoder(bytes.NewBuffer(b))
+	decoder.SetStrict(strict)
+
+	var outs []interface{}
+	for {
+		var tmp interface{}
+		err := decoder.Decode(&tmp)
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+		outs = append(outs, tmp)
+	}
+
+	if len(outs) == 0 {
+	} else if len(outs) == 1 {
+		// if there is only one value, we don't want to put it into a slice
+		item := outs[0]
+		// the we got null as the input
+		if item == nil {
+			outV.Set(reflect.Zero(outV.Type()))
+			return nil
+		}
+		outV.Set(reflect.ValueOf(item))
+	} else {
```

Comment:
handle `>0` here?

Reply:
adding this here `if len(outs) > 1 {` and removing the `if len(outs) == 0 {` branch.

Reply:
The `else` already is covering the ` > 1` in either case.

Reply:
if `if len(outs) == 0` is removed and if it remains a no-op, this branch has to handle `> 1`.